src/pyhf/infer/__init__.py

```diff
@@ -131,6 +132,16 @@ def hypotest(
     par_bounds = par_bounds or pdf.config.suggested_bounds()
     fixed_params = fixed_params or pdf.config.suggested_fixed()
 
+    if pdf.config.poi_index is None:
+        raise exceptions.UnspecifiedPOI(
+            'No POI is defined. A POI is required to run a hypothesis test.'
+        )
+
+    if not utils.all_pois_floating(pdf, fixed_params):
+        raise exceptions.InvalidModel(
+            f'POI at index [{pdf.config.poi_index}] forced to be fixed, cannot run inference'
```

Comment:
```suggestion
            f'POI at index [{pdf.config.poi_index}] is set as fixed, which makes inference impossible. Please unfix the POI to continue.'
```

Maybe we should be super explicit? There's probably some cleaner/clearer way then what this suggestion has, but I'm just trying to indicate that we should probably be overly verbose in error messages like this.